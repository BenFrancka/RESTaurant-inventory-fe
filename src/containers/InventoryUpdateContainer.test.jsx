import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import UpdateInventoryItemAmount from './InventoryUpdateContainer';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import item from '../fixtures/item.json';

const server = setupServer(
  rest.get('http://localhost:7890/api/v1/inventory/:id', (req, res, ctx) => {
    return res(ctx.json(item));
  })
);

describe('inventory update screen', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('displays current item', async () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/1']}>
        <UpdateInventoryItemAmount />
      </MemoryRouter>
    );

    screen.getByAltText('loading-spinner');

    const inventoryItem = await screen.findByRole('article', {
      name: 'inventory item',
    });
    expect(inventoryItem).not.toBeEmptyDOMElement();
    expect(container).toMatchSnapshot();
  });
});
