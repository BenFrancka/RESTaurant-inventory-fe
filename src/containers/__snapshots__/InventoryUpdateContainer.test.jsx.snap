// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inventory update screen displays current item 1`] = `
<div>
  <header>
    <h1>
      RESTaurant Inventory 
    </h1>
    <h3>
      <a
        href="/add"
      >
        Add an Item
      </a>
    </h3>
    <h3>
      <a
        href="/"
      >
        Return to Inventory
      </a>
    </h3>
  </header>
  <form>
    <label
      for="totalItems"
    >
      Number of Items: 
    </label>
    <input
      id="totalItems"
      name="totalItems"
      type="number"
      value=""
    />
    <label
      for="totalPrice"
    >
       Total Price: 
    </label>
    <input
      id="totalPrice"
      name="totalPrice"
      type="number"
      value=""
    />
  </form>
  <article
    aria-label="inventory item"
  >
    <h2 />
    <p>
      Category:
       
    </p>
    <p>
      Item Price:
       
    </p>
    <p>
      Total Items:
       
    </p>
    <p>
      Total Price:
       
    </p>
  </article>
</div>
`;
